src/main/java/org/trebol/jpa/services/predicates/ProductCategoriesPredicateJpaServiceImpl.java

```diff
@@ -58,46 +58,9 @@ public Predicate parseMap(Map<String, String> queryParamsMap) {
       String paramName = entry.getKey();
       String stringValue = entry.getValue();
       try {
-        switch (paramName) {
-          case "id":
-            return getBasePath().id.eq(Long.valueOf(stringValue));
-          case "code":
-            return getBasePath().code.eq(stringValue);
-          case "name":
-            predicate.and(getBasePath().name.eq(stringValue));
-            break;
-          case "nameLike":
-            predicate.and(getBasePath().name.likeIgnoreCase("%" + stringValue + "%"));
-            break;
-          case "parentCode":
-            if (stringValue != null && !stringValue.isEmpty()) {
-              predicate.and(getBasePath().parent.code.eq(stringValue));
-            }
-            break;
-          case "parentId":
-            if (stringValue == null || stringValue.isEmpty()) {
-              predicate.and(getBasePath().parent.isNull());
-            } else {
-              predicate.and(getBasePath().parent.id.eq(Long.valueOf(stringValue)));
-            }
-            break;
-          case "rootId":
-            if (stringValue != null && !stringValue.isEmpty()) {
-              List<Long> branchParentIds = treeResolver.getBranchIdsFromRootId(Long.valueOf(stringValue));
-              predicate.and(getBasePath().parent.id.in(branchParentIds));
-            }
-            break;
-          case "rootCode":
-            if (stringValue != null && !stringValue.isEmpty()) {
-              List<Long> branchParentIds = treeResolver.getBranchIdsFromRootCode(stringValue);
-              if (!branchParentIds.isEmpty()) {
-                predicate.and(getBasePath().parent.id.in(branchParentIds));
-              }
-            }
-            break;
-          default:
-            break;
-        }
+
+        PredicateCommand.getPredicate(paramName, stringValue, treeResolver, getBasePath(), predicate);
+
```

Comment:
This refactoring, on the surface hurts consistency, when you compare this service now, to other `PredicateJpaServiceImpl`s.

But besides (and most importantly), @NyorJa did you really feel that the `switch` block looked hard to understand, before the refactoring? Was it really confusing? I'm curious to know.

Reply:
I never thought we can tag it as false positive but I can revert it

Reply:
As I'm saying above, you can submit it to another PR, we can polish and tailor it further down.

Reply:
I will revert the refactored switch case